src/omni/embed.py

```diff
@@ -61,19 +61,24 @@ def __init__(
         embed_secret: str | None = None,
         vanity_domain: str | None = None,
     ):
-        _organization_name = organization_name or OmniEnv.ORGANIZATION_NAME
-        _vanity_domain = vanity_domain or OmniEnv.VANITY_DOMAIN
-        if not (_organization_name or _vanity_domain):
-            raise ValueError("'vanity_domain' or 'organization_name' are required.")
-        _embed_secret = embed_secret or OmniEnv.EMBED_SECRET
-        if not _embed_secret:
-            raise ValueError(
-                "embed_secret is required if it is not configured in environment variables."
+        omni_config = OmniConfig(
+            required_attrs=["embed_secret"],
+            organization_name=organization_name,
+            embed_secret=embed_secret,
+            vanity_domain=vanity_domain,
+        )
+        if not omni_config.vanity_domain and not omni_config.organization_name:
+            raise OmniConfigError(
+                "You must pass the vanity_domain or organization_name arguments OR "
+                "set the OMNI_ORGANIZATION_NAME or OMNI_VANITY_DOMAIN environment variables."
             )
-
-        embed_host = _vanity_domain or f"{_organization_name}.embed-omniapp.co"
+        embed_host = (
+            omni_config.vanity_domain
+            or f"{omni_config.organization_name}.embed-omniapp.co"
+        )
         self.embed_login_url = f"https://{embed_host}/embed/login"
-        self.embed_secret = _embed_secret
+        assert omni_config.embed_secret  # Required to appease mypy.
```

Comment:
To be consistent, we should probably `raise OmniConfigError` here if `not omni_config.embed_secret` -- pretty sure that would also appease mypy? Either way, we should raise a config error since you introduced that exception class rather than a generic assertion error

Reply:
The OmniConfigError will actually have been raised on line 65 since it is set as a required attr. Unfortunately mypy isn't able to detect that so I had to toss in an assert so the type is switched from `str | None` to `str`.

Reply:
I was going to do that but the logic is different for the API client. If you instantiated the client and were missing the configuration it would give you a weird error that you needed to set the vanity domain or organization name which wouldn't be true.

https://github.com/tillable/omni-sdk/pull/13/files#diff-fac7c92919555451551d837c35a10a763a887c9e9da5cfcd18de8aaf242ead31R19